pkgs/by-name/at/atlauncher/package.nix

```diff
@@ -1,50 +1,63 @@
-{ copyDesktopItems, fetchurl, jre, lib, makeDesktopItem, makeWrapper, stdenv, udev, xorg }:
+{ copyDesktopItems
+, fetchFromGitHub
+, fetchurl
+, lib
+, makeDesktopItem
+, makeWrapper
+, stdenv
 
-stdenv.mkDerivation (finalAttrs: {
+, # dependencies
+  jre
+, udev
+, xorg
+}:
+
+let
   pname = "atlauncher";
-  version = "3.4.35.4";
+  version = "3.4.35.5";
 
-  src = fetchurl {
-    url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${finalAttrs.version}/ATLauncher-${finalAttrs.version}.jar";
-    hash = "sha256-M8ygN70yizJM6VEffBh/lH/DneKAzQ5UFzc3g51dja0=";
+  srcs = {
+    source = (fetchFromGitHub {
+      owner = "ATLauncher";
+      repo = "ATLauncher";
+      rev = "v${version}";
+      sha256 = "sha256-9jOvElwb3jmO18MZOFqbB/mF75SdKufZvM1oWLKZDII=";
+    });
+    jar = (fetchurl {
+      url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${version}/ATLauncher-${version}.jar";
+      hash = "sha256-Y3Xe85Pte55ijIpJubOCfqIodSfpigJte2i/6eYzyO8=";
+    });
```

Comment:
Generally building from source is preferred, please remove this and set `src` to the source code fetched with `fetchFromGitHub`. Then you can access the packaging directory inside the repository and you won't need this second source.

Reply:
I'm aware that building from source is preferred. I've spent a lot of hours into trying to build this from source; but due to the lack of gradle packages in nixpkgs for reference and a general skill issue, I've not gotten a functioning from-source package.

Reply:
Fair enough. In that case I think you should keep the icon and .desktop
file set up as they were previously, for simplicity's sake. No need to
add extra sources if you don't have to.

Reply:
There already was an extra source for icon anyways. That too an unstable one (I spike to the upstream dev about it and they said it will be stable but you never know). I just changed that source to be GitHub and tagged the `.desktop` file along with it. Don't think this should be an issue.